WalletWasabi.Gui/Tabs/WalletManager/LoadWallets/LoadWalletViewModel.cs

```diff
@@ -312,5 +295,28 @@ private bool TrySetWalletStates()
 
 			return false;
 		}
+
+		#region IDisposable Support
+
+		private bool _disposedValue = false; // To detect redundant calls
+
+		protected virtual void Dispose(bool disposing)
```

Comment:
This will never get called?!

Reply:
It is called.

Reply:
Called here: https://github.com/molnard/WalletWasabi/blob/3dc0dfc9f906cf37eaf19f704680e19c1730990c/WalletWasabi.Gui/Tabs/WalletManager/WalletManagerViewModel.cs#L104

It turned out that WalletManager was there all the time even after it was closed. Now it disposes its items and recreates them on Open.